subprojects/file-collections/src/test/groovy/org/gradle/api/internal/file/DefaultFilePropertyFactoryTest.groovy

```diff
@@ -234,8 +234,8 @@ class DefaultFilePropertyFactoryTest extends Specification {
         tree.files
 
         then:
-        def e5 = thrown(IllegalStateException)
-        e5.message == 'Cannot query the value of this property because it has no value available.'
+        // Is that OK?
+        noExceptionThrown()
```

Comment:
This is what `FileTree` has to say about it:

```java
    /**
     * Returns the contents of this tree as a flattened Set.
     *
     * <p>The order of the files in a {@code FileTree} is not stable, even on a single computer.
     *
     * @return The files. Returns an empty set if this tree is empty.
     */
    @Override
    Set<File> getFiles();
```

Is empty the same as no value available? I don't know. It's definitely a behavior change.

To me the existing behavior makes sense, and I'd prefer not to change it, but I don't feel super-strong about this. Wonder what the implications are if we do change it...

Reply:
I changed it back so we throw when we call `directoryProperty.asFileTree.files`.